openquake/hazardlib/gsim/bozorgnia_campbell_2016.py

```diff
@@ -0,0 +1,465 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2014-2020 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`BozorgniaCampbell2016`
+               :class:`BozorgniaCampbell2016HighQ`
+               :class:`BozorgniaCampbell2016LowQ`
+               :class:`BozorgniaCampbell2016AveQJapanSite`
+               :class:`BozorgniaCampbell2016HighQJapanSite`
+               :class:`BozorgniaCampbell2016LowQJapanSite`
+"""
+import numpy as np
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib.gsim.campbell_bozorgnia_2014 import \
+                                          CampbellBozorgnia2014 as CB14
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+class BozorgniaCampbell2016(GMPE):
```

Comment:
If the functional form is the same used in CB14 wouldn't it be better to create this new model by extending the original class as in this example:

```
class BozorgniaCampbell2016(CB14):
```
In this way, you can avoid rewriting many of the methods as you did further down.

Reply:
Thanks for the suggestion, Marco. I would also agree with you that extending the CB14 class would give a better-looking code. I would like to note, however, that my reason for not doing that is by extending CB14, the new `BozorgniaCampbell2016` class would also get the now unnecessary methods (in particular, as is the general case in vertical-component GMPEs, the nonlinear functions for site response). I personally don't like this idea of inheriting unnecessary methods from CB14, so I just sacrificed aesthetics instead.

That being said, I would like to ask you and @micheles if there is a way that we can achieve BOTH CASES, i.e. not inheriting unnecessary methods from CB14 AND not having too many method calls from CB14.

Would appreciate further suggestions in light of the considerations above. Thanks!

Reply:
Clearly we can introduce a base class containing the essential methods and have both CB14 and BC15 inheriting from it, each one adding its own methods.

Reply:
Hi Francis, thanks for your thoughts. I agree with your strategy. Let's keep it this way for the time being. Thank you again for your contributions.

Reply:
LGTM